tests/components/sensor/test_tube_state.py

```diff
@@ -0,0 +1,46 @@
+"""The tests for the tube_state platform."""
+import unittest
+import requests_mock
+
+from homeassistant.components.sensor import tube_state
+from homeassistant.components.sensor.tube_state import CONF_LINE, URL
+from homeassistant.setup import setup_component
+from tests.common import load_fixture, get_test_home_assistant
+
+
+class TestLondonTubeSensor(unittest.TestCase):
+    """Test the tube_state platform."""
+
+    def add_entities(self, new_entities, update_before_add=False):
+        """Mock add entities."""
+        if update_before_add:
+            for entity in new_entities:
+                entity.update()
+
+        for entity in new_entities:
+            self.entities.append(entity)
+
+    def setUp(self):
+        """Initialize values for this testcase class."""
+        self.hass = get_test_home_assistant()
+        self.config = {CONF_LINE: ['London Overground']}
+        self.entities = []
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    def test_setup_with_config(self):
```

Comment:
Please merge this test with the one below. So instead of `tube_state.setup_platform(…)` just call `setup_component(…)`.

Then you can read the state from the state machine.

```python
state = self.hass.states.get('sensor.london_overground')
assert state.state == 'bla'
assert state.attributes.get('Description') == 'Something'
```